x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/RoleDescriptor.java

```diff
@@ -209,7 +257,8 @@ public boolean equals(Object o) {
         if (Arrays.equals(indicesPrivileges, that.indicesPrivileges) == false) return false;
         if (Arrays.equals(applicationPrivileges, that.applicationPrivileges) == false) return false;
         if (metadata.equals(that.getMetadata()) == false) return false;
-        return Arrays.equals(runAs, that.runAs);
+        if (Arrays.equals(runAs, that.runAs) == false) return false;
+        return Arrays.equals(remoteIndicesPrivileges, that.remoteIndicesPrivileges);
```

Comment:
Shouldn't null be effectively the same as empty array in the comparision?

Reply:
I'm going to make the field not nulleable so this shouldn't matter anymore.